etlman/projects/tests/test_views/test_view_script.py

```diff
@@ -90,3 +94,59 @@ def get_response_from_post_data_to_view(
             follow=True,
         )
         return response
+
+
+@pytest.mark.django_db
+class TestHTMLInViews:
+    def test_pipeline_authorizer(self, nonadmin_client):
+        pipeline = PipelineFactory()
+        response = nonadmin_client.get(
+            reverse("projects:pipeline_list", args=(pipeline.project.pk,)), follow=True
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN.numerator
+
+    def test_pipeline_table_headers_and_tags_in_html(
+        self, nonadmin_user, nonadmin_client
+    ):
+        pipeline = PipelineFactory()
+        CollaboratorFactory(project=pipeline.project, user=nonadmin_user)
+        response = nonadmin_client.get(
+            reverse("projects:pipeline_list", args=(pipeline.project.pk,)), follow=True
+        )
+        html = str(response.content)
+
+        assert response.status_code == HTTPStatus.OK.numerator
+        assert "Name" in html
+        assert "Actions" in html
+        assert "<thead>" in html
+        assert "<table" in html
+        assert "Add Pipeline" in html
+
+    def test_pipeline_table_content_in_html_one_pipeline(
+        self, nonadmin_user, nonadmin_client
+    ):
+        pipeline = PipelineFactory()
+        CollaboratorFactory(project=pipeline.project, user=nonadmin_user)
+        response = nonadmin_client.get(
+            reverse("projects:pipeline_list", args=(pipeline.project.pk,)), follow=True
+        )
+        html = str(response.content)
+
+        assert response.status_code == HTTPStatus.OK.numerator
+        assert pipeline.name in html
```

Comment:
Could you check the template context here instead of the HTML directly? Same in next test.

https://docs.djangoproject.com/en/4.1/topics/testing/tools/#django.test.Response.context